src/code42cli/cmds/alerts.py

```diff
@@ -242,21 +235,92 @@ def search(
     **kwargs,
 ):
     """Search for alerts."""
-    output_header = ext.try_get_default_header(
+    output_header = try_get_default_header(
         include_all, _get_default_output_header(), format
     )
     formatter = OutputFormatter(format, output_header)
     cursor = _get_alert_cursor_store(cli_state.profile.name) if use_checkpoint else None
-    handlers = ext.create_handlers(
-        cli_state.sdk,
-        AlertExtractor,
-        cursor,
-        use_checkpoint,
-        formatter=formatter,
-        force_pager=include_all,
-    )
-    _call_extractor(cli_state, handlers, begin, end, or_query, advanced_query, **kwargs)
-    handle_no_events(not handlers.TOTAL_EVENTS and not errors.ERRORED)
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        checkpoint = cursor.get(checkpoint_name)
+        if checkpoint is not None:
+            begin = checkpoint
+
+    query = _construct_query(cli_state, begin, end, advanced_query, or_query)
+    alerts_list = _get_all_alerts(cli_state, query)
+
+    if not alerts_list:
+        click.echo("No results found.")
+        return
+
+    if use_checkpoint:
+        checkpoint_name = use_checkpoint
+        # update checkpoint to alertId of last event retrieved
+        alerts_gen = _dedupe_checkpointed_events_and_store_updated_checkpoint(
+            cursor, checkpoint_name, alerts_list
+        )
+        formatter.echo_formatted_list(list(alerts_gen))
+    else:
+        formatter.echo_formatted_list(alerts_list)
+
+
+def _construct_query(state, begin, end, advanced_query, or_query):
+
+    if advanced_query:
+        state.search_filters = advanced_query
+    else:
+        if begin or end:
+            state.search_filters.append(
+                create_time_range_filter(f.DateObserved, begin, end)
+            )
+    if or_query:
+        state.search_filters = convert_to_or_query(state.search_filters)
+    verify_filter_groups(state.search_filters)
```

Comment:
I don't think this check is necessary anymore, since it's just our code that is constructing the filters. The extractor would get them passed in, so it was possible to get something that wasn't actually a filter group.